assertk/src/jvmMain/kotlin/assertk/assertions/any.kt

```diff
@@ -46,12 +46,13 @@ fun <T : Any> Assert<T>.doesNotHaveClass(jclass: Class<out T>) = given { actual
  * @see [isNotInstanceOf]
  * @see [hasClass]
  */
-fun <T : Any, S : T> Assert<T>.isInstanceOf(jclass: Class<S>): Assert<S> = transform { actual ->
+fun <T : Any> Assert<Any?>.isInstanceOf(jclass: Class<T>): Assert<T> = transform { actual ->
```

Comment:
Having two type params here was mostly useless. Kotlin will happily widen `T` to `Any` if the two have no common supertypes. The tests actually rely on this behavior.

Reply:
Think this was left over from an older version, looks like the common one was already change and this was missed